Bottom: 390a619f0e8aa1e599fca9d37e673bc4adcc46be
Top:    e670863f161c92e9816865d7b6831a9322a905b7
Author: Ian Kent <raven@themaw.net>
Date:   2013-08-19 13:45:03 +0800

Refresh of autofs-5.0.7-allow-use-of-hosts-map-in-maps.patch

---

diff --git a/modules/parse_sun.c b/modules/parse_sun.c
index 2026fc4..c138b71 100644
--- a/modules/parse_sun.c
+++ b/modules/parse_sun.c
@@ -1601,8 +1601,11 @@ int parse_mount(struct autofs_point *ap, const char *name,
 		}
 
 		/* if it's not a hosts map loc must be non-null */
-		if (!(strstr(options, "fstype=autofs") &&
+		if ((strstr(options, "fstype=autofs") &&
 		      strstr(options, "hosts"))) {
+			loc = NULL;
+			loclen = 0;
+		} else {
 			loclen = strlen(loc);
 			if (loclen == 0) {
 				free(loc);
